wordpress-mobile/WordPress-Android
# Stats revamp v2 action cards

This PR adds action cards to Insights view on Stats that appear when data meets certain criteria

Fixes #16581 
- [x] Depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2434)

| Reminder and Schedule | Grow |
|---|---|
|![Screenshot_20220521_224158](https://user-images.githubusercontent.com/990349/169679728-3187eeeb-2c96-44a9-ac79-efe972760940.png)| ![Screenshot_20220521_224232](https://user-images.githubusercontent.com/990349/169679746-6ec7a85b-e84e-4c41-9a11-1aabf12f70b6.png)| 


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app
- Launch / Re-Launch app
- Go to Stats either using quick links or menu 
- Switch to Insights tab if necessary
- Make sure **Most Popular Time** and **Total Followers** cards are visible on Insights tab. If those cards are not present then add them through stats management using either [+ add new stats card ] at the bottom or clicking ⚙️ icon at the top right hand corner.

Test 1: 

- A **Reminder action card** appears below **Most Popular Time** card 
- Click on the SET REMINDERS, it should take to blogging reminders flow on my site.
- Go through the blogging reminders setup and make sure setup works
- NOTE: The guide only appears if best day and hour has numbers

Test 2:
- For this a draft post is required.  Choose a site with a draft post or create one and return to Stats
- A **Schedule action card**  appears below **Most Popular Time** card if there are any drafts locally and reminder card is not showing (unlike in the image above, they do not appear together)
- Click on the SET SCHEDULE, it should take to DRAFTS tab on Posts view.
- NOTE: The guide only appears if there's a draft 

Test 3:
- A **Grow action card** as appears below **Total Followers** card 
- Click on the CHECK THE COURSE, it should open Followers page on WordPress.com support site.
- NOTE: The guide only appears if total followers <= 1

Test 4:
- Click on DISMISS on any or all of the cards and the card/s should disappear.  They wouldn't appear again for the site

NOTE:
- At the moment, action cards only appear once and do not appear again once dismissed or action taken on the card.

## Regression Notes
1. Potential unintended areas of impact
Ensure cards appear and disappear without affecting current cards 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated tests

3. What automated tests I added (or what prevented me from doing so)
Existing and updated tests

PR submission checklist:
